### PR TITLE
Extract ensure/rescue clause from method body if no begin block is present

### PR DESCRIPTION
**Pull request by st0012**
### Motivation

If the method body is a `PM_BEGIN_NODE` instead of a `PM_STATEMENTS_NODE`, it means the method definition doesn't have an explicit begin block. If that's the case, we need to check if the body contains an ensure or rescue clause, and if so, we need to elevate that node to the top level of the method definition, without the `Kwbegin` node to match the behavior of Sorbet's legacy parser.

### Test plan

See included automated tests.

### CODE REVIEW

File: test/prism_regression/ensure.rb [hunk 1]
```diff
@@ -7,6 +7,24 @@
   bar
 end
 
+# Testing an ensure clause in a method definition
+def method_with_ensure
+  method_1
+  method_2
+ensure
+  ensured_method
+end
```
Comment (amomchilov) [inline test/prism_regression/ensure.rb:16]: Can you also test the statements being empty? I think that might lead to `stmts` being `null`

```ruby
def method_with_ensure
ensure
  ensured_method
end
```

File: parser/prism/Translator.cc [hunk 2]
```diff
@@ -462,6 +462,21 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
...
+                // If that's the case, we need to check if the body contains an ensure or rescue clause, and if so,
+                // we need to elevate that node to the top level of the method definition, without the Kwbegin node to
+                // match the behavior of Sorbet's legacy parser.
+                auto kwbeginNode = dynamic_cast<parser::Kwbegin *>(body.get());
+
+                if (kwbeginNode != nullptr && kwbeginNode->stmts[0] != nullptr &&
+                    (dynamic_cast<parser::Rescue *>(kwbeginNode->stmts[0].get()) != nullptr ||
+                     dynamic_cast<parser::Ensure *>(kwbeginNode->stmts[0].get()) != nullptr)) {
+                    body = move(kwbeginNode->stmts[0]);
+                }
```
Comment (amomchilov) [inline parser/prism/Translator.cc:477]: This looks similar to the `inlineIfSingle` logic of `Translator::translateStatements()`. Is it possible to re-use that here somehow?

**Review by amomchilov — APPROVED** [review]

File: parser/prism/Translator.cc (hunk 2 above)
Comment (st0012) [reply parser/prism/Translator.cc:477]: I can't think of any case that it'd have multiple `Ensure` or `Rescue` or other statements. 

If both `Ensure` and `Rescue` are present, `Rescue` will be wrapped inside `Ensure` with the actual method body:

```
       Ensure {
          body = Rescue {
            body = Begin { # Actual method body
              stmts = [
                String {
                  val = <U string1>
                }
                String {
                  val = <U string2>
                }
              ]
            }
            rescue = [
              Resbody {
                exception = NULL
                var = NULL
                body = String {
                  val = <U rescued rescue>
                }
              }
            ]
            else_ = String {
              val = <U rescued else>
            }
          }
          ensure = String {
            val = <U ensure>
          }
        }
```

But I can add an unreachable statement to help us debug future issues in case there are exceptions.
